src/util/debug.c

```diff
@@ -292,15 +292,18 @@ void sss_vdebug_fn(const char *file,
          * searchable.
          */
         va_copy(ap_fallback, ap);
-        if (debug_chain_id > 0) {
+        if (debug_chain_id > 0 && debug_chain_id_fmt != NULL) {
+            strncpy(combined_fmt, debug_chain_id_fmt, sizeof(combined_fmt) - 1);
+            strcat(combined_fmt, "%s");
```

Comment:
This is actually not safe. Result of `strncpy()` might be non null-terminated or "-1" might be not enough to hold additonal "%s". See below.

Reply:
More important issue is that this `strncpy+strcat` will happen every DEBUG message with the same args every time - performance hit.

Reply:
`combined_fmt` should be static and initialized once.

We could also use just:
```
#define DEBUG_CHAIN_ID_FMT_RID "[RID#%lu] %s"
#define DEBUG_CHAIN_ID_FMT_CID "[CID#%lu] %s"
```
  --  You already make an assumption about specifiers inside `debug_chain_id_fmt` anyway.

But this would require `sss_debug_backtrace_printf(level, debug_chain_id_fmt, debug_chain_id, "");`...

Reply:
Okay thanks for the comments, I switched to using `snprintf` as proposed and `combined_fmt` is now static